synthesizer/program/src/resources/credits.aleo

```diff
@@ -106,556 +149,462 @@ record credits:
 
 /**********************************************************************************************************************/
 
-// This function allows any staker to bond their microcredits to a validator and specify a withdrawal address.
-// The corresponding functions for 'bond_public' are 'unbond_public' and 'claim_unbond_public'.
-function bond_public:
-    // Input the validator's address.
-    input r0 as address.public;
-    // Input the withdrawal address.
-    input r1 as address.public;
-    // Input the amount of microcredits to bond.
-    input r2 as u64.public;
+// bond_validator is used to initiall bond a validator into the active committee.
+// bond_validator can also be used to increase the self-bond of the validator.
+function bond_validator:
+    input r0 as address.public;  // Withdrawal address.
+    input r1 as u64.public;      // Amount of microcredits to bond.
+    input r2 as u8.public;       // Commission rate.
 
-    // Determine if the amount is at least one credit.
-    gte r2 1_000_000u64 into r3;
-    // Enforce the amount is at least one credit.
+    // Determine if the amount is at least 100 credits.
+    gte r1 100_000_000u64 into r3;
```

Comment:
Can you explain the rationale behind the choice of this number?

Can you also explain the implications of the choice of this number to the safety of the Bullshark BFT?

Reply:
I'll move it into the checks for a new validator entering the committee. It should only be 1 credit here as @raychu86 points out that, you should be able to increment by 1 credit with an initial self bond of 100 credits.

In terms of the safety of consensus, you should be able to set the minimum here to 0 theoretically. There should be no game theoretic difference between being delegating 10M credits and self bonding 10M credits, especially after slashing or rewards scaling is introduced. This is also extremely standard across dozens of PoS ecosystems according to @kyle-u410.

The reason 100 credits here is used is mostly practical. We could have approached this in multiple ways but instead of introducing more mappings or a new `remove_validator` function, I choose a minimum amount that a validator could practically unbond. It's a somewhat arbitrary but small amount that enables to a validator to reasonably estimate when they're trying to full unbond while not knowing the exact value onchain when the transaction is included in a block.

Reply:
The threshold to activate a validator remains at 10,000,000 credits. This change allows delegated credits to contribute to that minimum balance instead of requiring a validator to self bond.